### PR TITLE
[CLASS_TRANSFORMERS] Extends class-transform package with EnumFallback decorator

### LINKED ISSUE #834: [CLASS_TRANSFORMER] Allow to give fallback value to enum
# Description
Currently, it is possible to check that the value of a property is included in an enum definition using `isEnum` validator. Its behaviour is the following:
- If the value is one of the literal value of the enum, then the object is validated
- Otherwise, throw an error

Most of the time, this behaviour is desired but it may be too strict in other use cases. For those cases, having the ability to provide a fallback value seems to be a good solution. The behaviour would be:
- If the value is one of the literal value of the enum, then the value is kept
- Otherwise, return a default value of the enum

# Solution proposed
Creating a new transformer `EnumFallback` which allows to provide a fallback function and the enum type. Here is an example of its usage:
```ts
export enum UserRole {
  ADMIN = 'ADMIN',
  READER = 'READER',
}

class UserDto {
  @EnumFallback({
    type: UserRole,
    fallback: (value: UserRole) => UserRole.READER // if the role is not "ADMIN" or "READER", then the role will be "READER". 
  })
  public role?: UserRole;
}
```

This solution is possible according to [class transformer documentation](https://github.com/typestack/class-transformer?tab=readme-ov-file#additional-data-transformation). In addition, it would require to create a new library `@algoan/nestjs-class-transformer` as we did for class validators.

By the way, you should notice that if you use the transformer in conjunction with Nest validation and the `whitelist` option is enabled (see [documentation](https://docs.nestjs.com/techniques/validation#using-the-built-in-validationpipe)), you will have to add the decorator `@Allow` too:

```ts
class UserDto {
  @EnumFallback({
    type: UserRole,
    fallback: (value: UserRole) => UserRole.READER
  })
  @Allow() // required if whitelist = true
  public role?: UserRole;
}
```
If this decorator is missing, then the enum property will be simply removed from the returned object.

# FAQs
- **Why not providing a fallback function rather than just a value?**
As a developper, you will probably want to add logs when the fallback is called and have more complex logic in the fallback. Returning a value wouldn't allow this.
- **Why a transformer and not a validator?**
Please read [this comment of another issue](https://github.com/algoan/nestjs-components/issues/833#issuecomment-1850393855).

# References
Close #833

### PR DESCRIPTION
**Pull request by amarlankri**
## Description
Create class-transformers package with new decorator `EnumFallback`. This decorator can be used as follows:
```ts
import { EnumFallback } from '@algoan/nestjs-class-transformers';

export enum UserRole {
  ADMIN = 'ADMIN',
  READER = 'READER',
}

class User {
  @EnumFallback({
    type: UserRole,
    fallback: (value: UserRole) => UserRole.READER // if the role is not "ADMIN" or "READER", then the role will be "READER".
  })
  public role?: UserRole;
}
```

Finishes implementation started in #836

## Motivation and Context
Fix #834 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/class-transformers/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "@algoan/nestjs-class-transformers",
```
Comment (g-ongenae) [inline packages/class-transformers/package.json:2]: This file is incorrect. It lacks `publishConfig` and other properties. See [`class-validators/package.json`](https://github.com/algoan/nestjs-components/blob/master/packages/class-validators/package.json) for more details.

File: packages/class-transformers/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "version": "1.0.0",
+  "description": "",
```
Comment (g-ongenae) [inline packages/class-transformers/package.json:4]: Add a description here, it is the one that is used by GitHub packages.

**Review by g-ongenae — APPROVED** [review]

File: packages/class-transformers/src/enum-fallback.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  }, transformOptions);
+
+const transformEnumValue = <T>(value: T, { type, fallback }: EnumFallbackOptions<T>): T => {
+  // eslint-disable-next-line no-null/no-null
+  if (value === undefined || value === null) {
+    return value;
+  }
+
+  if (!Object.values(type).includes(value)) {
+    return fallback(value);
```
Comment (qhdinh) [inline packages/class-transformers/src/enum-fallback.decorator.ts:44]: If there is error in the fallback function, you think we should throw error or return a specific value?

File: packages/class-transformers/test/enum-fallback-decorator.test.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      @EnumFallback({ type: UserRole, fallback: () => UserRole.READER }, { groups: ['group1'] })
+      public role?: UserRole;
+    }
+
+    const user = plainToInstance(User, { role: 'WRITER' }, { groups: ['group2'] });
+
+    expect(user.role).toEqual('WRITER');
+  });
+
+  it('should return the fallback value for each invalid element if the property is an array', async () => {
```
Comment (qhdinh) [inline packages/class-transformers/test/enum-fallback-decorator.test.ts:67]: Could you please add a test when the fallback function is more than just returning a value? e.g. verifying a logging as we discussed

**Review by qhdinh — APPROVED** [review]

File: packages/class-transformers/src/enum-fallback.decorator.ts (hunk 3 above)
Comment (amarlankri) [reply packages/class-transformers/src/enum-fallback.decorator.ts:44]: I don't think so 🤔 Unlike the logging feature, this decorator may have a real impact on the application behaviour. In my opinion, it is more like the function you pass to `Array.map` for instance. It means that if the given function fails, it is not the responsibility to the library to handle it.

What do you think?

Comment (qhdinh) [reply packages/class-transformers/src/enum-fallback.decorator.ts:44]: Okay, so let the client handle it himself.

**Review by meriamBenSassi — APPROVED** [review]

**Pull request merged**